app/lib/features/onboarding/pages/analytics_opt_in_page.dart

```diff
@@ -26,61 +34,280 @@ class AnalyticsOptInPage extends ConsumerWidget {
+          decorationColor: Theme.of(context).colorScheme.primary,
+          decorationThickness: 2,
+          color: Theme.of(context).colorScheme.primary,
+        ),
+        textAlign: TextAlign.center,
+      ),
+    );
+  }
+
+  // Refactored crash analytics section with toggle functionality
+  Widget _buildCrashAnalytics(
+    BuildContext context,
+    WidgetRef ref,
+    TextTheme textTheme,
+  ) {
     final lang = L10n.of(context);
     final allowReportSending =
         ref.watch(allowSentryReportingProvider).valueOrNull ?? isNightly;
 
-    void toggle(bool? input) {
-      setCanReportToSentry(input ?? !allowReportSending);
-      ref.invalidate(allowSentryReportingProvider);
-    }
+    final matomoAnalyticsEnabled =
+        ref.watch(allowMatomoAnalyticsProvider).valueOrNull ?? false;
 
-    return ListTile(
-      onTap: () => toggle(null),
-      title: Text(lang.sendCrashReportsTitle),
-      subtitle: Text(lang.sendCrashReportsInfo),
-      leading: Switch(
-        value: allowReportSending,
-        onChanged: (value) => toggle(value),
-      ),
+    return Column(
+      children: [
+        FutureBuilder<bool>(
+          future: Future.wait([
+            _getAnalyticsPreference(basicTelemetryPref),
+            _getAnalyticsPreference(researchPref),
+          ]).then(
+            (values) =>
+                values.every((value) => value) &&
+                allowReportSending &&
+                matomoAnalyticsEnabled,
+          ),
+          builder: (context, snapshot) {
+            final allEnabled = snapshot.data ?? isNightly;
+            return _buildSwitchOption(lang.togglleAll, allEnabled, () async {
+              final newValue = !allEnabled;
+              await setCanReportToSentry(newValue);
+              await setMatomoAnalyticsEnabled(newValue);
+              await _setAnalyticsPreference(basicTelemetryPref, newValue);
+              await _setAnalyticsPreference(researchPref, newValue);
+              ref.invalidate(allowSentryReportingProvider);
+              ref.invalidate(allowMatomoAnalyticsProvider);
+            }, context);
+          },
+        ),
+        _buildCrashAnalyticsCard(
+          lang.sendCrashReportsTitle,
+          lang.sendCrashReportsInfo,
+          allowReportSending,
+          (value) async {
+            await setCanReportToSentry(value);
+            ref.invalidate(allowSentryReportingProvider);
+          },
+          context,
+          textTheme,
+        ),
+        _buildAnalyticsCard(
+          lang.basicTelemetry,
+          lang.basicTelemetryInfo,
+          basicTelemetryPref,
+          context,
+          textTheme,
+          ref,
+        ),
+        _buildCrashAnalyticsCard(
+          lang.appAnalytics,
+          lang.appAnalyticsInfo,
+          matomoAnalyticsEnabled,
+          (value) async {
+            await setMatomoAnalyticsEnabled(value);
+            ref.invalidate(allowMatomoAnalyticsProvider);
```

Comment:
using `ref.invalidate` is a strong indicator that the way this provider works should be re-evaluated. Because if this was a `ref.read(provider.notifier).set(newValue)` instead, it wouldn't need a `ref.invalidate` ...